### ISSUE TITLE
Beats & Measures: new dragging behaviour when snapping is turned on

### ISSUE THREAD

**Issue by Tantacrul** [audacity/audacity#4285]
### Your idea

When the user opts to turn on snapping, we require an additional behaviour: clips that are moved should _also_ snap to the selected unit of measurement. This is essential for Beats & Measures to function effectively. See the full [**Figma design here**](https://www.figma.com/file/SKDJypvy37EAauTPgwo2tL/07_WX---Beats-%26-measures?node-id=568%3A17189&t=UKmXqBrifEeSaFAe-1)

![image](https://user-images.githubusercontent.com/21022311/217015608-020cbb37-c83a-48db-9d74-60feb60c8772.png)

### Problem to be solved

This change will allow musical creation work to be much easier and predictable. It is an essential component of adding Beats and Measures functionality.

**Comment by petersampsonaudacity** [comment]
Testing this on W10 with @crsib 's branch build for this:  https://github.com/audacity/audacity/actions/runs/4574323691

This works as @Tantacrul  intends for dragging clips when snapping is "on"

However it does _**NOT**_ affect the behaviour for**Tracks > Align Tracks** - is this intended desired behaviour?

Note that this was always the case for **Align Tracks** - see this clip from the 3.2.5 Manual:
![image](https://user-images.githubusercontent.com/50205984/229117358-23e6883b-34b3-492c-99cd-6ff39082f95f.png)

On this page:
https://manual.audacityteam.org/man/selection_toolbar.html

**Comment by petersampsonaudacity** [comment]
I ask this mainly in the interests of getting the manual's documentation correct.

**Comment by Tantacrul** [comment]
Musically speaking, it's not massively useful for align tracks to ignore snapping. I just tried out a few scenarios and found this:

- With snapping to a beat (or time value) on, snapping behaved as expected
- If I wanted to snap to a clip on another track, I was able to by switching the 'snap' checkbox off

It's hard to make a very confident judgement here, since it's really about user expectation. However, the way it is working in the build you referenced makes sense to me. It reduces the possibility of accidentally misaligning things.

I'd suggest we see how this performs and potentially create an option for the user to control snapping more finely in preferences.
